vendor/github.com/metal3-io/baremetal-operator/pkg/apis/metal3/v1alpha1/baremetalhost_types.go

```diff
@@ -475,6 +475,18 @@ func (cs CredentialsStatus) Match(secret corev1.Secret) bool {
 	return true
 }
 
+// RebootMode defines known variations of reboot modes
+type RebootMode string
+
+const (
+	RebootModeHard RebootMode = "hard"
+	RebootModeSoft RebootMode = "soft"
+)
+
+type RebootAnnotationArguments struct {
+	Mode RebootMode `json:"mode"`
+}
+
```

Comment:
Looks like this was added manually. This will be overridden as soon as someone will run `go mod vendor`
You first need backport this from metal3/bmo to openshift/bmo and later update the dependency [here](https://github.com/openshift/cluster-api-provider-baremetal/blob/master/go.mod#L22)

Reply:
You're right, am working on the PR so I can reference this properly. Thanks!

Reply:
https://github.com/openshift/baremetal-operator/pull/128